ihmc-high-level-behaviors/src/main/java/us/ihmc/behaviors/behaviorTree/BehaviorTreeRootNodeExecutor.java

```diff
@@ -154,8 +170,6 @@ private void executeNextAction()
       state.getLogger().info("Triggering action execution: %s".formatted(actionToExecute.getDefinition().getName()));
       actionToExecute.update();
       actionToExecute.triggerActionExecution();
-      actionToExecute.updateCurrentlyExecuting();
-      currentlyExecutingActions.add(actionToExecute);
```

Comment:
Removing these two lines is the most significant change in the PR. Since updateCurrentlyExecuting can set executing to false, there was a possibility of inconsistent and undefined state. Now it's only called in one place, making catching success and failure events guaranteed.

Reply:
I think my thinking was to call these here to keep state consistent, but that ruins other state guarantees. It's better to call it on the next update so it's only done in one place.